### ISSUE TITLE
unable tail logs

### ISSUE THREAD

**Issue by sunadm** [docker/docker#15199]
docker version:
Client:
 Version:      1.8.0-rc1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   4f1c66a
 Built:        Sat Jul 25 05:54:28 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.0-rc1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   4f1c66a
 Built:        Sat Jul 25 05:54:28 UTC 2015
 OS/Arch:      linux/amd64

docker info:

Containers: 1
Images: 41
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 45
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-59-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 2
Total Memory: 1.955 GiB
Name: aliyun
ID: 4F2J:TRGP:BCNT:WP5C:E62W:QV3B:6GBZ:A7B7:F7OX:5DBG:H7R6:2DMK
WARNING: No swap limit support

uname -a:

Linux aliyun 3.13.0-59-generic #98-Ubuntu SMP Fri Jul 24 21:05:26 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Provide additional environment details (AWS, VirtualBox, physical, etc.):

List the steps to reproduce the issue:
1. execute command "docker logs -f --tail 100 container_name"
2.
3.

Describe the results you received:
It output some logs and then break back to shell prompt.

Describe the results you expected:
the logs output continuous.

**Comment by cpuguy83** [comment]
Logs are only followed if the container is running.
Is your container running?

**Comment by sunadm** [comment]
yes, of course 




    _____________________________

**Comment by cpuguy83** [comment]
Is there an error on the daemon?

**Comment by sunadm** [comment]
ok, got something.

```
INFO[12778] GET /v1.20/containers/json?all=1             
INFO[12787] GET /v1.20/containers/json?all=1             
INFO[12787] GET /v1.20/containers/a7b029a1b06c4b3c4f6edea66609dc3c57dd4ecf06aceff18efc03666286e9eb/json 
INFO[12788] GET /v1.20/containers/dianjiadistribution_registry_1/json 
INFO[12788] GET /v1.20/containers/dianjiadistribution_registry_1/logs?follow=1&stderr=1&stdout=1&tail=100 
ERRO[12788] Error streaming logs: json: cannot unmarshal number into Go value of type jsonlog.JSONLog
```

**Comment by calavera** [comment]
I can reproduce this with a simple `docker run -d alpine ping google.com` and then running that logs command.

**Comment by cpuguy83** [comment]
I see the issue, working on a fix.

**Comment by cpuguy83** [comment]
Fix is in #15209
